ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,376 @@
+    @objc open var expandedHeightFraction: CGFloat = 1.0 {
+        didSet {
+            if expandedHeightFraction != oldValue && !respectsPreferredContentSize {
+                updateBottomSheetHeightConstraints()
+            }
+        }
+    }
+
+    /// Height of the top portion of the content view that should be visible when the bottom sheet is collapsed.
+    @objc open var collapsedContentHeight: CGFloat = Constants.defaultCollapsedContentHeight
+
+    /// The object that acts as the delegate of the bottom sheet.
+    @objc open weak var delegate: BottomSheetControllerDelegate?
+
+    // MARK: - View loading
+
+    public override func loadView() {
+        view = BottomSheetPassthroughView()
+
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(bottomSheetView)
+
+        bottomSheetView.addGestureRecognizer(panGestureRecognizer)
+        panGestureRecognizer.delegate = self
+
+        NSLayoutConstraint.activate([
+            bottomSheetView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            bottomSheetView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            bottomSheetOffsetConstraint
+        ])
+    }
+
+    private lazy var resizingHandleView: ResizingHandleView = {
+        let resizingHandleView = ResizingHandleView()
+        resizingHandleView.isAccessibilityElement = true
+        resizingHandleView.accessibilityTraits = .button
+        resizingHandleView.isUserInteractionEnabled = true
+        resizingHandleView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleResizingHandleViewTap)))
+        return resizingHandleView
+    }()
+
+    private lazy var bottomSheetView: BottomSheetView = {
+        let bottomSheetView = BottomSheetView()
+        bottomSheetView.translatesAutoresizingMaskIntoConstraints = false
+
+        let stackView = UIStackView(arrangedSubviews: [resizingHandleView, contentView])
+        stackView.spacing = 0.0
+        stackView.axis = .vertical
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        bottomSheetView.contentView.addSubview(stackView)
+
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: bottomSheetView.topAnchor),
+            stackView.leadingAnchor.constraint(equalTo: bottomSheetView.leadingAnchor),
+            stackView.trailingAnchor.constraint(equalTo: bottomSheetView.trailingAnchor),
+            stackView.bottomAnchor.constraint(equalTo: bottomSheetView.bottomAnchor, constant: -Constants.Spring.overflowHeight)
+        ])
+
+        return bottomSheetView
+    }()
+
+    // MARK: - Gesture handling
+
+    @objc private func handleResizingHandleViewTap(_ sender: UITapGestureRecognizer) {
+        if currentOffsetFromBottom != collapsedOffsetFromBottom {
+            animate(to: .collapsed, velocity: 0)
+            hostedScrollView?.setContentOffset(.zero, animated: true)
```

Comment:
does this animation happen concurrently with the animate(to:, velocity:) call? Do we need to sync or stage them?

Reply:
It is concurrent. The scroll view just scrolls all the way up while the bottom sheet is collapsing. I don't think we need to do anything more than this.

Reply:
Becuase we're using property animations there's no risk of a layout pass/commit causing an in-progress animation to snap to finish when the first one finishes, right?